lib/timers.js

```diff
@@ -409,6 +393,26 @@ function processImmediate() {
   }
 }
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw) {
+      if (!L.isEmpty(queue)) {
+        // Handle any remaining on next tick, assuming we're still
+        // alive to do so.
+        while (!L.isEmpty(immediateQueue)) {
+          L.append(queue, L.shift(immediateQueue));
+        }
+        immediateQueue = queue;
+        process.nextTick(processImmediate);
+      }
```

Comment:
I guess I was wondering if the looping within a `finally` isn't hit by a deopt.

If we instead returned that it threw, and moved it into `processImmediate()` would it be more efficient?